corehq/apps/app_manager/tests/test_practice_user_config.py

```diff
@@ -0,0 +1,129 @@
+        delete_all_users()
+        cls.domain = "practice-user-domain"
+
+        cls.factory = AppFactory(build_version='2.28.0', domain=cls.domain)
+        module, form = cls.factory.new_basic_module('register', 'case')
+        form.source = cls.get_xml('very_simple_form')
+
+        cls.project = create_domain(cls.domain)
+        cls.user = CommCareUser.create(cls.domain, 'test@main-domain.commcarehq.org', 'secret')
+
+    def assert_(self, app, expected_paths):
+        # test user-restore files created
+        # test resource blocks created
+        app.create_build_files
+
+    def tearDown(self):
+        turn_off_demo_mode(self.user)
+
+    @staticmethod
+    def _get_restore_resource(version, build_profile_id=None):
+        extra = "?profile={id}".format(id=build_profile_id) if build_profile_id else ""
+        return """
+        <partial>
+        <user-restore>
+           <resource id="practice-user-restore" version="{version}" descriptor="practice user restore">
+             <location authority="local">./practice_user_restore.xml</location>
+             <location authority="remote">./practice_user_restore.xml{extra}</location>
+           </resource>
+        </user-restore>
+        </partial>
+        """.format(version=version, extra=extra)
+
+    def test_app_specific(self):
+        turn_on_demo_mode(self.user, self.domain)
+        app = self.factory.app
+        app.practice_mobile_worker_id = self.user._id
+
+        # check suit contains user-restore resource
+        self.assertXmlPartialEqual(
+            self._get_restore_resource(self.user.demo_restore_id),
+            app.create_suite(),
+            "./user-restore"
+        )
+        # check 'files/practice_user_restore.xml' is included in the build files
+        app.create_build_files(save=True)
+        app.save()
+        self.assertTrue(app.lazy_fetch_attachment('files/practice_user_restore.xml'))
+
+    def test_profile_specific(self):
+        turn_on_demo_mode(self.user, self.domain)
+        app = self.factory.app
+        build_profile_id = "some_uuid"
+        app.build_profiles[build_profile_id] = BuildProfile(
+            langs=['en'], name='en-profile', practice_mobile_worker_id=self.user._id
+        )
+
+        self.assertXmlPartialEqual(
+            self._get_restore_resource(self.user.demo_restore_id, build_profile_id),
+            app.create_suite(build_profile_id=build_profile_id),
+            "./user-restore"
+        )
+        app.create_build_files(save=True, build_profile_id=build_profile_id)
+        app.save()
+        self.assertTrue(app.lazy_fetch_attachment('files/practice_user_restore.xml'))
+
+    def test_bad_config(self):
+        # if the user set as practice user for an app is not practice user, build should raise error
+
+        # refetch so that memoized app.get_practice_user gets busted
+        app = Application.get(self.factory.app._id)
```

Comment:
This is dependent on some of the other tests running first in order to save the app. Rather create a new app from scratch.